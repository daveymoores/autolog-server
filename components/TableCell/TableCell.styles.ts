import css from "styled-jsx/css";

export default css`
  .timesheet__table--cell {
    display: flex;
    align-items: center;
    justify-content: center;
    height: var(--cellHeight);
    width: var(--cellHeight);
  }

  @media (prefers-color-scheme: dark) {
    .timesheet__table--cell {
      background-color: var(--darkGrey);
    }

    .timesheet__table--cell-hour {
      background-color: var(--mistyGrey);
    }

    .timesheet__table--cell-weekend {
      background-color: var(--lightGreen);
      color: var(--darkGrey);
    }
  }

  @media (prefers-color-scheme: light) {
    .timesheet__table--cell {
      background-color: var(--lightGreen);
    }

    .timesheet__table--cell-hour {
      background-color: var(--mistyGreen);
    }

    .timesheet__table--cell-weekend {
      background-color: var(--darkGrey);
      color: var(--lightGreen);
    }
  }
`;
