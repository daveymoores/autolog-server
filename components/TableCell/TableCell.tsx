import React from "react";

import { ParsedTimesheetDayLog } from "../../pages/[timesheet]";
import styles from "./TableCell.styles";

interface TableCellProps
  extends Partial<Pick<ParsedTimesheetDayLog, "weekend" | "hours">> {
  day?: boolean;
  day_index?: number;
}

const TableCell: React.FC<TableCellProps> = ({
  hours,
  weekend,
  day,
  day_index,
}) => {
  return (
    <>
      {day ? (
        <div className="timesheet__table--cell">{day_index}</div>
      ) : (
        <div
          className={`timesheet__table--cell${
            weekend ? " timesheet__table--cell-weekend" : ""
          }`}
        >
          {hours}
        </div>
      )}
      <style jsx>{styles}</style>
    </>
  );
};

export default TableCell;
