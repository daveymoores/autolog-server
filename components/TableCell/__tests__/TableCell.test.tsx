import React from "react";
import renderer from "react-test-renderer";

import TableCell from "../TableCell";

describe("TableCell", () => {
  it("renders TableCell unchanged", () => {
    const tree = renderer.create(<TableCell />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
