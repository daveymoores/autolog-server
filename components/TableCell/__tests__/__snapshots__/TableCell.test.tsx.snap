// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCell renders TableCell unchanged 1`] = `
<div
  className="jsx-2108230735 timesheet__table--cell"
/>
`;
